lib/nexthop.c

```diff
@@ -36,6 +36,63 @@
 DEFINE_MTYPE_STATIC(LIB, NEXTHOP, "Nexthop")
 DEFINE_MTYPE_STATIC(LIB, NH_LABEL, "Nexthop label")
 
+int nexthop_cmp(const struct nexthop *next1, const struct nexthop *next2)
+{
+	int ret;
+	uint32_t n1, n2;
+
+	if (next1->vrf_id < next2->vrf_id)
```

Comment:
Is this restricted to {-1, 0, 1} ? if not, can we just subtract the vrf_ids?

Reply:
You mean this?
```
if (next1->vrf_id - next2->vrf_id)
     return (next1->vrf_id - next2->vrf_id)
```
If you think that is better style, I will change it.

Reply:
I think the explicit `{-1, 0, 1}` is preferable, because it's easy to fail to consider the consequences of overflow.  While that might not matter in this case, consistent style helps prevent other cases.

Performance-wise - CPU cycles are mostly free.  Memory access is costly.  Capping it to `{-1, 0, 1}` is the former, so I'd refuse performance arguments :smile:.

Reply:
I'm fine with David's preference - just wanted to ask the question.

Reply:
so ... I don't think I'm making a "performance argument" - I'm just pointing out that we're doing some special stuff to get {-1, 0, 1} in some paths, but then falling through to `memcmp()`, which does ... arithmetic. it's ok with me, personally, but I wouldn't want someone to be tempted to rely on anything other than {<0, 0, >0} in the result.